src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -624,12 +624,11 @@ private static void CheckParameterModifiers(BaseParameterSyntax parameter, Bindi
                 hasErrors = true;
             }
             else if (conversion.IsReference &&
-                (parameterType.SpecialType == SpecialType.System_Object || parameterType.Kind == SymbolKind.DynamicType) &&
```

Comment:
Have you gone back in the history to find out why this line was added? This seems extremely deliberate, so I want to have a good understanding of why it was there in the first place.

Reply:
This line dates from Pilchie's "Hello World" commit.
I should have copied the info here from the bug thread, but the native compiler correctly produced an error:
`test.cs(6,37): error CS1763: 'x' is of type 'System.Collections.Generic.IEnumerable<char>'. A default parameter value of a reference type other than string can only be initialized with null`

Reply:
The whole block is from the `Roslyn.Says("Hello World");` commit. Is there somewhere internal that should be searched as well?

Reply:
> This line dates from Pilchie's "Hello World" commit.

I guessed so, but I think this is something I would like information from TFS on. Sorry :slightly_smiling_face:.

Reply:
Ok, so @jcouv and I did some spelunking through many layers of TFS to find that this was added by Eric at least a decade ago. Unless @ericlippert remembers why this originally only checked if the parameter type was `object` and wants to chime in here, I'd say I'm good with the change, _provided_ VS still builds. This is one I think we want a validation PR for.

Reply:
What is the native compiler behavior?

Reply:
> What is the native compiler behavior?

@AlekseyTs Julien tested it in the bug itself. It was an error: https://github.com/dotnet/roslyn/issues/59789#issuecomment-1051745067

Reply:
Good spelunking. I definitely wrote that line of code, and I have no memory of why I added that extra check. I concur that it is unnecessary.